libmscore/instrument.cpp

```diff
@@ -1181,6 +1181,42 @@ bool Instrument::operator==(const Instrument& i) const
          &&  i._singleNoteDynamics == _singleNoteDynamics;
       }
 
+//---------------------------------------------------------
+//   operator!=
+//---------------------------------------------------------
+
+bool Instrument::operator!=(const Instrument& i) const
+      {
+      int n = _longNames.size();
+      if (i._longNames.size() != n)
+            return true;
+      for (int k = 0; k < n; ++k) {
+            if (!(i._longNames[k] == _longNames[k]))
+                  return true;
+            }
+      n = _shortNames.size();
+      if (i._shortNames.size() != n)
+            return true;
+      for (int k = 0; k < n; ++k) {
+            if (!(i._shortNames[k] == _shortNames[k].name()))
+                  return true;
+            }
+
+      return i._minPitchA != _minPitchA
+            || i._maxPitchA != _maxPitchA
+            || i._minPitchP != _minPitchP
+            || i._maxPitchP != _maxPitchP
+            || i._useDrumset != _useDrumset
+            || i._midiActions != _midiActions
+            || i._articulation != _articulation
+            || i._transpose.diatonic != _transpose.diatonic
+            || i._transpose.chromatic != _transpose.chromatic
+            || i._trackName != _trackName
+            || !(*i.stringData() == *stringData())
+            || i._singleNoteDynamics != _singleNoteDynamics;
+      //return !operator==(i);
+      }
```

Comment:
Why not just make this whole function return `!(*this == i)`? (That syntax may not be exactly correct off the top of my head).

Reply:
I do see now that that's been commented out. Why?

Reply:
This was something I was unsure about. Because the equals function compares the channels vector, and creating a new instrument in an instrument change creates a new channel, it would not return equal, even if they were the same instrument, which is what I was wanting to check. However, thinking about it, this should probably be in a new function, as it's quite confusing to have these work slightly differently.

Reply:
OK, that makes sense. As you say, I'd consider putting the conditions that are common between them into a new function and using that to reduce duplication.

Reply:
And make sure to comment why it's like that as well for confused people like me!